lib/dal/src/attribute/prototype/debug.rs

```diff
@@ -0,0 +1,207 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+use thiserror::Error;
+
+use crate::attribute::prototype::{
+    argument::{
+        static_value::StaticArgumentValue,
+        value_source::{ValueSource, ValueSourceError},
+        AttributePrototypeArgument, AttributePrototypeArgumentError,
+    },
+    AttributePrototypeError,
+};
+use crate::attribute::value::{AttributeValueError, ValueIsFor};
+
+use crate::prop::{PropError, PropPath};
+
+use crate::func::execution::{FuncExecution, FuncExecutionError};
+use crate::socket::input::InputSocketError;
+use crate::socket::output::OutputSocketError;
+use crate::workspace_snapshot::node_weight::NodeWeightError;
+use crate::workspace_snapshot::WorkspaceSnapshotError;
+use crate::{
+    AttributePrototype, AttributePrototypeId, AttributeValue, AttributeValueId, Component,
+    ComponentError, DalContext, Func, FuncError, FuncId, InputSocket, Prop,
+};
+use serde_json::Value;
+use telemetry::prelude::*;
```

Comment:
We should clean up these imports. Let's put all `use <non-crate>` statements at the top, then a blank line, then all `use crate` statements.

Some folks split `use std` statements in their files too, but we usually group them together (unclear if by habit or intention, but that's a topic for another day).

Reply:
We don't have a linter that will sort those for us?

Reply:
IDEs typically do. There are a few features needed to do this well via automation and at least one of them is not `stable` yet: https://rust-lang.github.io/rustfmt/?version=master&search=group#imports_granularity

Reply:
will do! yeah my IDE does some of this formatting and merging for me but I wasn't sure what 'more right' vs 'less right' looked like :)